### ISSUE TITLE
Numpy 2.0.0 breaks things

### ISSUE THREAD

**Issue by wd15** [usnistgov/fipy#1049]
Running with Numpy 2.0.0 breaks things. See [fipy.log](https://github.com/user-attachments/files/15874209/fipy.log).

Test failures like this:

```
   Traceback (most recent call last):                                                                                                                                                         
      File "/nix/store/ffy6s77zhixlwvzprz5j0zj4ma8n8h2q-python3-3.11.8/lib/python3.11/doctest.py", line 1355, in __run                                                                         
        exec(compile(example.source, filename, "single",                                                                                                                                       
...                                                                                             
      File "/nix/store/4xbcd56id9azry5srn5fwh83qlin9lgb-python3-3.11.8-env/lib/python3.11/site-packages/fipy/variables/meshVariable.py", line 99, in __init__                                  
        dtype = numerix.obj2sctype(value)                                                                                                                                                      
                ^^^^^^^^^^^^^^^^^^                                                                                                                                                             
    AttributeError: module 'fipy.tools.numerix' has no attribute 'obj2sctype'
```

Versions are as follows:

python      3.11.8 (main, Feb  6 2024, 21:21:21) [GCC 12.3.0]
fipy        3.4.4
numpy       2.0.0
pysparse    not installed
scipy       1.13.1
matplotlib  3.9.0

**Comment by guyer** [comment]
`numpy.obj2sctype` was [deprecated in numpy 2.0.0](https://github.com/numpy/numpy/blob/a4cddb60489f821a1a4dffc16cd5c69755d43bdb/numpy/_expired_attrs_2_0.py#L48). It looks like its original usage of converting to numpy's internal data types was [deprecated in numpy 1.20.0](https://numpy.org/doc/stable/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated). 

Unfortunately, the proposed replacement, `np.dtype(obj).type`, cannot be used as FiPy used `numpy.obj2sctype`, e.g., with numpy 1.24.3:
```python
>>> np.obj2sctype(np.array([1.0, 2.0]))
numpy.float64
```
but, with numpy 1.24.3 or 2.0.0,
```python
>>> np.dtype(np.array([1.0, 2.0])).type
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Cannot construct a dtype from an array
```

```python
>>> np.array([1.0, 2.0]).dtype
dtype('float64')
```
works with either, but I'm not sure if that covers all the cases where we used it.

**Comment by guyer** [comment]
I think I've fixed these issues in #1050
